source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown

```diff
@@ -0,0 +1,284 @@
+to the options the library providing. Here is the library `daru-view` uses.
+Please check the examples options, they are passing in javascript code:
+
+  * GoogleCharts: [https://developers.google.com/chart/interactive/docs/gallery](https://developers.google.com/chart/interactive/docs/gallery)
+
+  * HighCharts: [https://www.highcharts.com/demo](https://www.highcharts.com/demo)
+
+  * Nyaplot: [https://github.com/SciRuby/nyaplot](https://github.com/SciRuby/nyaplot) (it works same as `daru`)
+
+
+Note: User must have some knowledge about the plotting tool(that you want to
+use) to use it in `daru-view`. So that you can pass the correct options.
+
+
+### GoogleCharts:
+
+Set the plotting library to `:googlecharts` to use this adapter. This will
+load the required js files in your webpage or IRuby notebook.
+
+```ruby
+require 'daru/view'
+Daru::View.plotting_library = :googlecharts
+```
+
+Let's create a DataFrame :
+
+```
+idx = Daru::Index.new ['Year', 'Sales']
+data_rows = [
+          ['2004',  1000],
+          ['2005',  1170],
+          ['2006',  660],
+          ['2007',  1030]
+]
+df_sale_exp = Daru::DataFrame.rows(data_rows)
+df_sale_exp.vectors = idx
+```
+
+do some manupulation using `daru` if you want to do. Now time to plot it:
+
+```ruby
+line_basic_chart = Daru::View::Plot.new(df_sale_exp)
+line_basic_chart.chart
+```
+
+This will return the chart object we created using GoogleCharts.
+In IRuby notebook, you will like this:
+
+{% img https://github.com/Shekharrajak/gsoc_2017_blog/blob/master/img/googlecharts_line1.png?raw=true 'Baisc line chart using GoogleCharts' 'Baisc line chart using GoogleCharts' %}
+
+You can find the IRuby notebook example in [this link](http://nbviewer.jupyter.org/github/shekharrajak/daru-view/blob/master/spec/dummy_iruby/Google%20Chart%20%7C%20%20Line%20Chart.ipynb).
+
+These are various charts type we can use e.g. line, area, bar, bubble,
+candlestick, combo, histogram, org, pie, stepped area chart, timeline, treemap,
+gauge, column, scatter, etc. We can find the customization options in the
+[google charts site](https://developers.google.com/chart/interactive/docs/gallery).
+
+Let me try another chart type Geo :
+
+```ruby
+country_population = [
+          ['Germany', 200],
+          ['United States', 300],
+          ['Brazil', 400],
+          ['Canada', 500],
+          ['France', 600],
+          ['RU', 700]
+]
+
+df_cp = Daru::DataFrame.rows(country_population)
+df_cp.vectors = Daru::Index.new(['Country', 'Population'])
+
+geochart = Daru::View::Plot.new(
+    df_cp, type: :geo, adapter: :googlecharts
+)
+
+```
+
+
+Note: If you have already loaded the dependent JS files for the library then you can use `adpater: :googlecharts` in your Plot initialization.
+
+
+{% img https://github.com/Shekharrajak/gsoc_2017_blog/blob/master/img/googlechart_geo1.png?raw=true 'Baisc Geo chart using GoogleCharts' 'Baisc Geo chart using GoogleCharts' %}
+
```

Comment:
Maybe add small examples of each plotting library supported?

Reply:
I have added examples for Highcharts and Nyaplot as well. May be I should create different blog for details examples.